certora/specs/Health.spec

```diff
@@ -75,6 +78,44 @@ filtered {
     assert !priceChanged => stillHealthy;
 }
 
+// The general case of liquidate for the stayHealthy rule
+//   - the user checked is not the borrower,
+//   - the market of the liquidation is the market of the user,
+//   - there is still some borrow on the market after liquidation,
+//   - there was no bad debt realization.
+rule stayHealthyLiquidate(env e, MorphoHarness.MarketParams marketParams, address borrower, uint256 seizedAssets, bytes data) {
+    MorphoHarness.Id id = libId(marketParams);
+    address user;
+
+    priceChanged = false;
+    // Assume that the position is healthy before the interaction.
+    require isHealthy(marketParams, user);
+    uint256 debtSharesBefore = borrowShares(id, user);
+    uint256 debtAssetsBefore = summaryMulDivUp(debtSharesBefore, virtualTotalBorrowAssets(id), virtualTotalBorrowShares(id));
+    // Safe require because of the invariants onlyEnabledLltv and lltvSmallerThanWad in ConsistentState.spec.
+    require marketParams.lltv < 10^18;
+    // Assumption to ensure that no interest is accumulated.
+    require lastUpdate(id) == e.block.timestamp;
+
+    liquidate(e, marketParams, borrower, seizedAssets, 0, data);
+    require !priceChanged;
+
+    // Safe require because of the invariant sumBorrowSharesCorrect.
+    require borrowShares(id, user) <= totalBorrowShares(id);
+    // Handle the case where not everything was repaid.
+    require totalBorrowAssets(id) > 0;
+    // Handle the case where the user has some collateral left.
+    require collateral(id, borrower) > 0;
```

Comment:
I don't understand why we need to assume this? the account is obviously not healthy if it has some debt but not collateral

Reply:
This is the same thing as requiring that there is no bad debt realization, but it should be said more clearly